### ISSUE TITLE
Register EventPlugins in @Template

### ISSUE THREAD

**Issue by mhevery** [angular/angular#1187]
Currently the plugins for Event handling are registered in [`application.js`](https://github.com/angular/angular/blob/514ba5428201e9721eeb34e14e8329fcb2ef11be/modules/angular2/src/core/application.js#L90). Instead it should be register in `@Template`

```
@Component(...)
@Template({
  eventHandlers: [new MyEventHandler()]
})
class MyComponent{}

class MyEventHandler implements EventManagerPlugin {
}
```

* [ ] `EventHandler` should not deal with the `Zone`. Instead the all calls to `addEventListener` should be done outside the `Zone`. That way we need to cross the zone boundary only once per `View` creation. The callback function should than re-enter the `Zone` as part of the dispatch.
* [ ] rename `EventManagerPlugin` to `EventHandler`
* [ ] Also remove [`manager`](https://github.com/angular/angular/blob/514ba5428201e9721eeb34e14e8329fcb2ef11be/modules/angular2/src/core/events/event_manager.js#L47) from the supper type. This was only needed to get a hold of the Zone.
* [ ] `EventHandler` are instantiated before the compilation, and are singletons. 
* [ ] Question: Should DOM and Hammer be default / implied?

**Comment by tbosch** [comment]
Mmh, how does this work with the render layer? EventPlugins depend on the
DOM...

**Comment by mhevery** [comment]
Let's punt on this for a while

**Comment by mhevery** [comment]
This is obsolete, and no longer relevant / actionable. To keep our issues clean, we are aggressively closing them.
